superset-frontend/src/cccs-viz/plugins/plugin-chart-ajax/src/plugin/buildQuery.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { buildQueryContext, QueryFormData } from '@superset-ui/core';
+
+/**
+ * The buildQuery function is used to create an instance of QueryContext that's
+ * sent to the chart data endpoint. In addition to containing information of which
+ * datasource to use, it specifies the type (e.g. full payload, samples, query) and
+ * format (e.g. CSV or JSON) of the result and whether or not to force refresh the data from
+ * the datasource as opposed to using a cached copy of the data, if available.
+ *
+ * More importantly though, QueryContext contains a property `queries`, which is an array of
+ * QueryObjects specifying individual data requests to be made. A QueryObject specifies which
+ * columns, metrics and filters, among others, to use during the query. Usually it will be enough
+ * to specify just one query based on the baseQueryObject, but for some more advanced use cases
+ * it is possible to define post processing operations in the QueryObject, or multiple queries
+ * if a viz needs multiple different result sets.
+ */
+export default function buildQuery(formData: QueryFormData) {
+  /*
+    We receive an ip as a filter, our job is to find everthing there is to know about that ip
+    We fire multiple queries to multiple data sets and collect the results here.
+    */
+
+  const formDataCopy = {
+    ...formData,
+    result_type: 'post_processed',
+  };
+
+  return buildQueryContext(formDataCopy, baseQueryObject => {
+    // RAW mode (not aggregated)
+    // eslint-disable-next-line no-param-reassign
+    return [
+      {
+        ...baseQueryObject,
+        row_limit: 10,
```

Comment:
This is the row limit set for the iframe plugin, I'm not sure what I should change it to otherwise.

Reply:
I don't think 10 was chosen for a particular reason. The plugin checks how many rows were returned in the result and gives an error if it there is more than 1. For that reason i don't believe it matters whether the limit is 2 or 10 it is only there to make sure that the user selected one row because it only supports fetching one file